src/services/ghac/backend.rs

```diff
@@ -0,0 +1,555 @@
+        let p = build_abs_path(&self.root, path);
+
+        let url = format!("{}{CACHE_URL_BASE}/caches", self.cache_url);
+
+        let bs = serde_json::to_vec(&GhacReserveRequest {
+            key: p,
+            version: self.version.to_string(),
+            cache_size: size,
+        })
+        .map_err(parse_json_serialize_error)?;
+
+        let mut req = Request::post(&url);
+        req = req.header(AUTHORIZATION, format!("Bearer {}", self.catch_token));
+        req = req.header(ACCEPT, CACHE_HEADER_ACCEPT);
+        req = req.header(CONTENT_LENGTH, bs.len());
+        req = req.header(CONTENT_TYPE, "application/json");
+
+        let req = req
+            .body(AsyncBody::Bytes(Bytes::from(bs)))
+            .map_err(new_request_build_error)?;
+
+        Ok(req)
+    }
+
+    async fn ghac_upload(
+        &self,
+        cache_id: i64,
+        size: u64,
+        body: AsyncBody,
+    ) -> Result<Request<AsyncBody>> {
+        let url = format!("{}{CACHE_URL_BASE}/caches/{cache_id}", self.cache_url);
+
+        let mut req = Request::patch(&url);
+        req = req.header(AUTHORIZATION, format!("Bearer {}", self.catch_token));
+        req = req.header(ACCEPT, CACHE_HEADER_ACCEPT);
+        req = req.header(CONTENT_LENGTH, size);
+        req = req.header(CONTENT_TYPE, "application/octet-stream");
+        req = req.header(
+            CONTENT_RANGE,
+            BytesContentRange::default()
+                .with_range(0, size - 1)
+                .to_header(),
+        );
+
+        let req = req.body(body).map_err(new_request_build_error)?;
+
+        Ok(req)
+    }
+
+    async fn ghac_commmit(&self, cache_id: i64, size: u64) -> Result<Request<AsyncBody>> {
+        let url = format!("{}{CACHE_URL_BASE}/caches/{cache_id}", self.cache_url);
+
+        let bs =
+            serde_json::to_vec(&GhacCommitRequest { size }).map_err(parse_json_serialize_error)?;
+
+        let mut req = Request::post(&url);
+        req = req.header(AUTHORIZATION, format!("Bearer {}", self.catch_token));
+        req = req.header(ACCEPT, CACHE_HEADER_ACCEPT);
+        req = req.header(CONTENT_TYPE, "application/json");
+        req = req.header(CONTENT_LENGTH, bs.len());
+
+        let req = req
+            .body(AsyncBody::Bytes(Bytes::from(bs)))
+            .map_err(new_request_build_error)?;
+
+        Ok(req)
+    }
+
+    async fn ghac_delete(&self, path: &str) -> Result<Response<IncomingAsyncBody>> {
+        let p = build_abs_path(&self.root, path);
+
+        let url = format!(
+            "{}/repos/{}/actions/caches?key={}",
+            self.api_url,
+            self.repo,
+            percent_encode_path(&p)
+        );
+
+        let mut req = Request::delete(&url);
+        req = req.header(AUTHORIZATION, format!("Bearer {}", self.api_token));
+        req = req.header(
+            USER_AGENT,
+            format!(
+                "opendal {}/ghac",
+                env::var("CARGO_PKG_VERSION").unwrap_or_default()
```

Comment:
Use the [`env!`](https://doc.rust-lang.org/stable/std/macro.env.html) macro here. `CARGO_PKG_VERSION` will not be available at run time.

Reply:
Thanks a lot!

Reply:
Are you willing to send a PR for this?

Reply:
Fixed in https://github.com/datafuselabs/opendal/pull/1125